azure/scope/identity.go

```diff
@@ -66,6 +67,11 @@ func NewAzureCredentialsProvider(ctx context.Context, kubeClient client.Client,
 		return nil, errors.New("AzureClusterIdentity is not of type Service Principal")
 	}
 
+	if !IsClusterNamespaceAllowed(ctx, kubeClient, identity.Spec.AllowedNamespaces, azureCluster.Namespace) {
+		conditions.MarkFalse(azureCluster, infrav1.NetworkInfrastructureReadyCondition, infrav1.NamespaceNotAllowedByIdentity, clusterv1.ConditionSeverityError, "")
```

Comment:
I feel this will create an unexpected side effect that when instantiating a new credentials provider. I don't have any particularly good solution to it other than moving it to the controller. wdyt?

Reply:
what is the side effect? the condition

I was going to put in the controller first; but was trying to avoid doing an extra get call, 

I'll take another look at it

Reply:
Yeah, the conditions. One thing I did for handling [vm deletion error](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/8ebf049dd162a74a579ac7d25db726f8238dea86/controllers/azuremachine_controller.go#L275) was to return a new error type and used that to set the appropriate condition.

Reply:
are you concerned that the `condition` is set in the scope package? or that we are setting a condition?

cause the I did it that way after seeing the machine scope set conditions as well

sorry for all the questions, trying to understand

Reply:
Sorry about that bikeshedding. My thinking is that `NewAzureCredentialsProvider` should be just about creating a new CredentialProvider instance with the inputs passed. Since conditions kind of represent the state of an AzureCluster, I just thought it might not be the right place to set it. This is just an opinionated view. I'm fine with it remaining as is if you think otherwise.

Reply:
moved it to controller before creating scope, PTAL